spec/services/users/reset_progress_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+RSpec.describe Users::ResetProgress do
+  let(:user) { create(:user) }
+  let!(:foundations_path) { create(:path, default_path: true) }
+  let!(:foundations_course) { create(:course, path: foundations_path) }
+  let!(:foundations_section) { create(:section, course: foundations_course) }
+  let!(:foundations_lesson_one) { create(:lesson, section: foundations_section) }
+  let!(:foundations_lesson_two) { create(:lesson, section: foundations_section) }
+  let!(:foundations_lesson_one_completion) { create(:lesson_completion, lesson: foundations_lesson_one, user: user) }
+  let!(:foundations_lesson_two_completion) { create(:lesson_completion, lesson: foundations_lesson_two, user: user) }
+  subject(:service) { described_class.call(user) }
+
+  describe '#call' do
```

Comment:
I think we can move some things around in this test to avoid having to create all the rails records in the hen path is not the default path context.

Since we remove the lesson completions regardless of the user current path, we only need to test the users path is updated in the contexts:

```
describe '#call' do

  it 'deletes all lesson completions' do
    expect { service }.to change { user.lesson_completions.count }.from(2).to(0)
  end

  context 'when users path is the default path' do
    let(:path) { create(:path, default: true) }

    it 'does not change path' do
      expect { service }.not_to(change { user.path })
    end
  end

  context "when the users path is not the default path" do
    let(:path) { create(:path, default: false) }

    it 'does not change path' do
      expect { service }.not_to(change { user.path })
    end
  end

end
```

We would need to make the user let accept a dynamic let variable value for this to work 
`let(:user) { create(:user, path: path) }`

Reply:
This is a great idea, but I ran into two issues. 

First, I need both paths to exist for the test when it resets the path from rails to foundations. Otherwise, `Path.default_path` returns `nil`. 

Secondly, I do not think the factory is set up to be able to specify a path when the user is created. 

However, I will take this suggestion and only test the users path in the contexts. 👍